lib/galaxy/schema/visualization.py

```diff
@@ -100,3 +108,251 @@ class VisualizationSummaryList(RootModel):
+    )
+    href: str = Field(
+        ...,
+        title="Href",
+        description="The href of the plugin.",
+    )
+
+
+class VisualizationShowResponse(Model, WithModelClass):
+    model_class: VISUALIZATION_MODEL_CLASS = ModelClassField(VISUALIZATION_MODEL_CLASS)
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="ID",
+        description="Encoded ID of the Visualization.",
+    )
+    title: str = Field(
+        ...,
+        title="Title",
+        description="The name of the visualization.",
+    )
+    type: str = Field(
+        ...,
+        title="Type",
+        description="The type of the visualization.",
+    )
+    user_id: EncodedDatabaseIdField = Field(
+        ...,
+        title="User ID",
+        description="The ID of the user owning this Visualization.",
+    )
+    dbkey: Optional[str] = Field(
+        None,
+        title="DbKey",
+        description="The database key of the visualization.",
+    )
+    slug: Optional[str] = Field(
+        None,
+        title="Slug",
+        description="The slug of the visualization.",
+    )
+    latest_revision: VisualizationRevisionResponse = Field(
+        ...,
+        title="Latest Revision",
+        description="The latest revision of this Visualization.",
+    )
+    revisions: List[EncodedDatabaseIdField] = Field(
+        ...,
+        title="Revisions",
+        description="A list of encoded IDs of the revisions of this Visualization.",
+    )
+    url: str = Field(
+        ...,
+        title="URL",
+        description="The URL of the visualization.",
+    )
+    username: str = Field(
+        ...,
+        title="Username",
+        description="The name of the user owning this Visualization.",
+    )
+    email_hash: str = Field(
+        ...,
+        title="Email Hash",
+        description="The hash of the email of the user owning this Visualization.",
+    )
+    tags: Optional[TagCollection] = Field(
+        None,
+        title="Tags",
+        description="A list of tags to add to this item.",
+    )
+    annotation: Optional[str] = Field(
+        None,
+        title="Annotation",
+        description="The annotation of this Visualization.",
+    )
+    plugin: Optional[VisualizationPluginResponse] = Field(
+        None,
+        title="Plugin",
+        description="The plugin of this Visualization.",
+    )
+
+
+class VisualizationCreateResponse(Model):
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="ID",
+        description="Encoded ID of the Visualization.",
+    )
+
+
+class VisualizationUpdateResponse(Model):
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="ID",
+        description="Encoded ID of the Visualization.",
+    )
+    revision: EncodedDatabaseIdField = Field(
+        ...,
+        title="Revision",
+        description="Encoded ID of the Visualization Revision.",
+    )
+
+
+class VisualizationCreatePayload(Model):
+    type: Optional[SanitizedString] = Field(
```

Comment:
This probably has a set value of things that can be filled in, we should document and restrict those values.

Reply:
I couldn't find the specific set of values. Can you point me where to find them?

Reply:
I guess these are the plugin types loaded into VisualizationsRegistry. Might be a little tricky to load them up since you need the fully loaded plugins registry.

Reply:
Yeah, but make it a normal string. It must match the plugin type, no need to do sanitization there.

Reply:
just for `type`?

Reply:
yes